R/validate.R

```diff
@@ -0,0 +1,25 @@
+#' Validate triangle
+#' @description
+#' Various checks to make sure that the reporting triangle passed in to
+#' `estimate_delay()` is formatted properly.
+#' @param triangle a dataframe formatted as a reporting triangle
+#'
+#' @returns
+validate_triangle <- function(triangle) {
+  # check column names includes reference date
+  checkmate::assert_names(names(triangle), must.include = "reference_date")
+  # check that the columns are integer digits
+  checkmate::assert_integerish(
+    as.numeric(grep("^\\d+$", names(triangle), value = TRUE))
```

Comment:
put a nolint hash on it ideally targetted at the specific lint that is throwing a false positive

Reply:
I think checkmate is a pretty reasonable dependency that is quite common. Happy if you want to write your own but I wouldn't

Reply:
Ok I will use checkmate going forward, the ones I wrote so far didn't really require it